### PR TITLE
Seanshi/finetuning file format details

### PR DESCRIPTION
**Pull request by seanshi-scale**

### CODE REVIEW

File: docs/guides/fine_tuning.md [hunk 1]
```diff
@@ -1,3 +1,20 @@
-# Fine-Tuning
+# Fine Tuning API
+The Fine Tuning API allows you to fine tune various open-source LLMs on your own data, then make inference calls to the resulting LLM.
```
Comment (yixu34) [inline docs/guides/fine_tuning.md:2]: I think `open source` is more canonical? At least from what I've seen.

File: docs/guides/fine_tuning.md [hunk 2]
```diff
@@ -1,3 +1,20 @@
...
-Coming soon!
+## Preparing Data
+Your data must be formatted as a CSV file that includes two columns: `prompt` and `response`. The data needs to be uploaded to somewhere publicly accessible, so that we can read the data to fine tune on it. For example, you can upload your data to a public s3 bucket.
+
+## Launching the Fine Tune
+Once you have uploaded your data, you can use our API to launch a Fine Tune. You will need to provide the base model to train off of, the locations of the training and validation files, an optional set of hyperparameters to override, and an optional suffix to append to the name of the fine tune. 
+
+Currently, we support the following base models:
+
+- `mpt-7b-instruct`
```
Comment (yixu34) [inline docs/guides/fine_tuning.md:12]: Might be worth linking to the Model Zoo page (you can do something like `[Model Zoo](/model_zoo)` to do a markdown link to it). That way we don't have to worry about things getting out of sync.

File: docs/guides/fine_tuning.md [hunk 3]
```diff
@@ -1,3 +1,20 @@
...
+
+- `llama-7b`
+
+Once the fine tune is launched, you can also get the status of your fine tune.
+
+## Making inference calls to your fine tune
+
+Once the fine tune is finished, you will be able to start making requests to the model. First, you can list the LLMs that you can make inference requests to in order to get the name of your fine tuned model. You can then use that name to direct your inference requests.
```
Comment (yixu34) [inline docs/guides/fine_tuning.md:20]: Maybe remove the `that you can make inference requests to` part? It's assumed that any models that you own are able to receive requests.

Comment (yixu34) [reply docs/guides/fine_tuning.md:20]: Might also be good to link to the completions API for more details.

File: docs/guides/fine_tuning.md [hunk 4]
```diff
@@ -1,3 +1,20 @@
-# Fine-Tuning
+# Fine Tuning API
```
Comment (yixu34) [inline docs/guides/fine_tuning.md:1]: Might be good to link to the Fine tuning API reference somewhere up here to remind people where the detailed docs are.

File: clients/python/llmengine/client.py [hunk 5]
```diff
@@ -452,9 +452,15 @@ async def create_fine_tune_job(
 
         Args:
             training_file (`str`):
-                Path to file of training dataset
+                Path to file of training dataset.
```
Comment (yixu34) [inline clients/python/llmengine/client.py:455]: Might want to write these a bit more formally, i.e. with definite articles or 'this'.

**Review by yixu34 — COMMENTED** [review]
Nice! Had some comments

File: clients/python/llmengine/client.py (hunk 5 above)
Comment (seanshi-scale) [reply clients/python/llmengine/client.py]: oh wait also we're gonna have to update this client to pick up the field name changes

**Review by yixu34 — APPROVED** [review]

**Comment by yixu34** [comment]
Linking https://app.shortcut.com/scaleai/story/782393 - this gets us part of the way, we still need to figure out suggestions on file sizes ((like the training dataset format, max rows, best practices, etc)

**Pull request merged**